app/models/course.rb

```diff
@@ -53,6 +52,8 @@ class Course < ActiveRecord::Base
                                     dependent: :destroy
   has_many :lesson_plan_events, through: :lesson_plan_items,
                                 source: :actable, source_type: Course::LessonPlan::Event.name
+  # Achievements must be declared after material_folders or duplication will fail.
+  has_many :achievements, dependent: :destroy
```

Comment:
It affects the SQL insertion order and thus when the ID numbers are generated. Some models need ID numbers of the objects they depend on to save correctly.